python/ray/experimental/serve/tests/test_queue.py

```diff
@@ -0,0 +1,72 @@
+import ray
+from ray.experimental.serve.queues import CentralizedQueues
+
+
+def test_single_prod_cons_queue(serve_instance):
+    q = CentralizedQueues()
+    q.link("svc", "backend")
+
+    result_object_id = q.enqueue_request("svc", 1)
+    work_object_id = q.dequeue_request("backend")
+    got_work = ray.get(ray.ObjectID(work_object_id))
+    assert got_work.request_body == 1
+
+    ray.worker.global_worker.put_object(got_work.result_object_id, 2)
+    assert ray.get(ray.ObjectID(result_object_id)) == 2
+
+
+def test_alter_backend(serve_instance):
+    q = CentralizedQueues()
+
+    result_object_id = q.enqueue_request("svc", 1)
+    work_object_id = q.dequeue_request("backend-1")
+    q.set_traffic("svc", {"backend-1": 1})
+    got_work = ray.get(ray.ObjectID(work_object_id))
+    assert got_work.request_body == 1
+    ray.worker.global_worker.put_object(got_work.result_object_id, 2)
+    assert ray.get(ray.ObjectID(result_object_id)) == 2
+
+    result_object_id = q.enqueue_request("svc", 1)
+    work_object_id = q.dequeue_request("backend-2")
+    q.set_traffic("svc", {"backend-2": 1})
+    got_work = ray.get(ray.ObjectID(work_object_id))
+    assert got_work.request_body == 1
+    ray.worker.global_worker.put_object(got_work.result_object_id, 2)
+    assert ray.get(ray.ObjectID(result_object_id)) == 2
+
+
+def test_split_traffic(serve_instance):
+    q = CentralizedQueues()
+
+    q.enqueue_request("svc", 1)
+    q.enqueue_request("svc", 1)
+    q.set_traffic("svc", {})
+    work_object_id_1 = q.dequeue_request("backend-1")
+    work_object_id_2 = q.dequeue_request("backend-2")
+    q.set_traffic("svc", {"backend-1": 0.5, "backend-2": 0.5})
+
+    got_work = ray.get(
+        [ray.ObjectID(work_object_id_1),
+         ray.ObjectID(work_object_id_2)])
+    assert [g.request_body for g in got_work] == [1, 1]
```

Comment:
Is this guaranteed to pass? Isn't it probabilistic?

Reply:
Yes in current assumption. This will pass all the time. Because we are only running probabilities estimate when workers are fetching the query, if worker_1 fetch the query, the second query will definitely go to worker_2. This is _incorrect_ in the sense that we are not guaranteeing a perfect conformance to traffic policy. 

This will be fixed in a followup PR.